PacketDotNet/TcpPacket.cs

```diff
@@ -421,10 +421,29 @@ public TcpPacket(ByteArraySegment bas)
                 // the length of the payload is the total payload length
                 // above, minus the length of the tcp header
                 payloadPacketOrData.TheByteArraySegment.Length = newTcpPayloadLength;
+                this.DecodePayload();
             }
         }
 
         /// <summary>
+        /// Decode Payload to Support Drda procotol
```

Comment:
Can we move the changes to add decoding to TcpPacket into another commit so we can discuss them separately?